lib/nostrum/api.ex

```diff
@@ -778,6 +831,149 @@ defmodule Nostrum.Api do
     |> bangify
   end
 
+  @doc ~S"""
+  Gets a list of emojis for a given guild.
+
+  If the request is successful, this function returns `{:ok, emojis}`, where 
+  `emojis` is a list of `Nostrum.Struct.Emoji`. Otherwise, returns `{:error, reason}`.
+  """
+  @spec list_guild_emojis(Guild.id) :: error | {:ok, [Emoji.t]}
+  def list_guild_emojis(guild_id) do
+    request(:get, Constants.guild_emojis(guild_id))
+    |> handle_request_with_decode({:list, {:struct, Emoji}})
+  end
+
+  @doc ~S"""
+  Same as `list_guild_emojis/1`, but raises `Nostrum.Error.ApiError` in case of failure.
+  """
+  @spec list_guild_emojis!(Guild.id) :: no_return | [Emoji.t]
+  def list_guild_emojis!(guild_id) do
+    list_guild_emojis(guild_id)
+    |> bangify
+  end
+
+  @doc ~S"""
+  Gets an emoji for the given guild and emoji ids.
+
+  If the request is successful, this function returns `{:ok, emoji}`, where 
+  `emoji` is a `Nostrum.Struct.Emoji`. Otherwise, returns `{:error, reason}`.
+  """
+  @spec get_guild_emoji(Guild.id, Emoji.id) :: error | {:ok, Emoji.t}
+  def get_guild_emoji(guild_id, emoji_id) do
+    request(:get, Constants.guild_emoji(guild_id, emoji_id))
+    |> handle_request_with_decode({:struct, Emoji})
+  end
+
+  @doc ~S"""
+  Same as `get_guild_emoji/2`, but raises `Nostrum.Error.ApiError` in case of failure.
+  """
+  @spec get_guild_emoji!(Guild.id, Emoji.id) :: no_return | Emoji.t
+  def get_guild_emoji!(guild_id, emoji_id) do
+    get_guild_emoji(guild_id, emoji_id)
+    |> bangify
+  end
+
+  @doc ~S"""
+  Creates a new emoji for the given guild.
+
+  ## Options 
+ 
+    * `:name` (string) - name of the emoji
+    * `:image` (base64 data URI) - the 128x128 emoji image. Maximum size of 256kb
+    * `:roles` (list of `t:Nostrum.Struct.Snowflake.t/0`) - roles for which this emoji will be whitelisted
+    (default: [])
+
+  `:name` and `:image` are always required.
+
+  ## Return Values
```

Comment:
I've noticed that when there are options like you've included this sub-section. I don't think I'm a big fan of having it be its own section, but I'm open for discussion. We should decide how we want to display this, and make sure it's homogeneous throughout the module.

The elixir docs are kind of unorganized in how they do this, probably because their functions are a lot less linear than ours.

Reply:
Do you think it'd be okay to move under the original doc definition? I just feel it's out of place since every method has results (while not every method has optional params), but the section is only included on those methods that have optional params. I feel like that could be confusing?

A lot of times I point to the elixir docs just to see how they do it. [Here](https://hexdocs.pm/elixir/Enum.html#flat_map/2) for example they fit it in pretty smoothly.
While at the same time from the last pr we have [places](https://github.com/Kraigie/nostrum/blob/master/lib/nostrum/api.ex#L1409) that miss returning the value altogether.

I think it might be worth it to hash out a design or some sort of template (maybe through the discord channel for quicker iteration) we could use to kind of bring all of these docs in line. And then going forward we can follow that and I'll go through and fix up the previous PR if need be.